DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/RevitConverterToSpeckle.cs

```diff
@@ -29,24 +28,43 @@ ParameterValueExtractor parameterValueExtractor
   // if it cannot be converted then we should throw
   public Base Convert(object target)
   {
-    var objectConverter = _toSpeckle.ResolveInstance(target.GetType().Name);
+    var objectConverter = GetConversionForObject(target.GetType());
 
     if (objectConverter == null)
     {
-      throw new NotSupportedException($"No conversion found for {target.GetType().Name}");
+      throw new SpeckleConversionException($"No conversion found for {target.GetType().Name}");
     }
 
     Base result =
       objectConverter.Convert(target)
       ?? throw new SpeckleConversionException($"Conversion of object with type {target.GetType()} returned null");
 
     // POC : where should logic common to most objects go?
-    if (target is Element element)
+    // shouldn't target ALWAYS be DB.Element?
+    if (target is DB.Element element)
     {
       _convertedObjectsCache.AddConvertedBase(element.UniqueId, result);
       _parameterValueExtractor.RemoveUniqueId(element.UniqueId);
+
+      // POC: is this the right place?
+      result.applicationId = element.UniqueId;
     }
 
     return result;
   }
+
+  private IHostObjectToSpeckleConversion? GetConversionForObject(Type objectType)
+  {
+    if (objectType == typeof(object))
+    {
+      return null;
+    }
+
+    if (_toSpeckle.ResolveInstance(objectType.Name) is IHostObjectToSpeckleConversion conversion)
+    {
+      return conversion;
+    }
+
+    return GetConversionForObject(objectType.BaseType);
+  }
```

Comment:
I swear this was added by Connor the other day, or did you move it here?
If so I must have missed the deletion diff

Reply:
It was, I needed it for some reason I cannot recall, so he threw it my way